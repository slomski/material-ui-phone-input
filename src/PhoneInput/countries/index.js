import countryNames from './countries.json';

const countries = [];

countryNames.forEach(country => {
  countries.push(country);
});

export default countries;
