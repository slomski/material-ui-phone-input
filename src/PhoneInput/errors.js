const errors = {
  TOO_SHORT: 'Number is too short',
  TOO_LONG: 'Number is too long',
  INVALID_COUNTRY: 'Country code in number is invalid',
  NOT_A_NUMBER: 'Invalid number provided'
};

export default errors;
