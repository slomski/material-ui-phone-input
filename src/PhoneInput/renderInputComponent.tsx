import React from 'react';
import Autosuggest from 'react-autosuggest';
import TextField from '@material-ui/core/TextField';

export function renderInputComponent(inputProps: Autosuggest.InputProps<any>): React.ReactNode {
  const { classes, inputRef = () => {}, ref, onChange, value, ...rest } = inputProps;
  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement | HTMLSelectElement>) => {
    onChange(
      event as React.FormEvent<any>,
      { newValue: event.target.value, method: 'type' } as Autosuggest.ChangeEvent
    );
  };

  return (
    <TextField
      variant="filled"
      fullWidth
      InputProps={{
        inputRef: node => {
          ref(node);
          inputRef(node);
        },
        classes: {
          input: classes.input,
        },
      }}
      {...rest}
      onChange={handleChange}
      defaultValue={value}
    />
  );
}
