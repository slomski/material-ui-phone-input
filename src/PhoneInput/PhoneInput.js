import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField from '@material-ui/core/TextField';
import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
import Paper from '@material-ui/core/Paper';
import MenuItem from '@material-ui/core/MenuItem';
import Autosuggest from 'react-autosuggest';
import deburr from 'lodash/deburr';
import match from 'autosuggest-highlight/match';
import parse from 'autosuggest-highlight/parse';
import Popper from '@material-ui/core/Popper';
import { parsePhoneNumber } from 'libphonenumber-js';
import Fade from '@material-ui/core/Fade';
import errors from './errors';
import countries from './countries';
import FlagIcon from '../FlagIcon';

const suggestionsLists = countries.map(option => ({
  value: option.code,
  label: option.name,
  dialCode: option.dialCode
}));

const styles = () => ({
  suggestion: {
    display: 'block'
  },
  suggestionsList: {
    margin: 0,
    padding: 0,
    listStyleType: 'none'
  },
  input: {
    padding: 10,
    border: 0,
    width: '100%'
  }
});

function renderInputComponent(inputProps) {
  const { classes, inputRef = () => {}, ref, ...other } = inputProps;

  return (
    <TextField
      variant="filled"
      fullWidth
      InputProps={{
        inputRef: node => {
          ref(node);
          inputRef(node);
        },
        classes: {
          input: classes.input
        }
      }}
      {...other}
    />
  );
}

function getSuggestions(country) {
  const inputValue = deburr(country.trim()).toLowerCase();
  const inputLength = inputValue.length;
  let count = 0;

  return inputLength === 0
    ? []
    : suggestionsLists.filter(suggestion => {
        const keep = count < 5 && suggestion.label.slice(0, inputLength).toLowerCase() === inputValue;
        if (keep) {
          count += 1;
        }
        return keep;
      });
}

function renderSuggestion(suggestion, { query, isHighlighted }) {
  const matches = match(suggestion.label, query);
  const parts = parse(suggestion.label, matches);

  return (
    <MenuItem selected={isHighlighted} component="div">
      <div>
        {parts.map((part, index) =>
          part.highlight ? (
            <span key={String(index)} style={{ fontWeight: 500 }}>
              {part.text}
            </span>
          ) : (
            <strong key={String(index)} style={{ fontWeight: 400 }}>
              {part.text}
            </strong>
          )
        )}
      </div>
    </MenuItem>
  );
}

function getSuggestionValue(suggestion) {
  return suggestion.label;
}

class PhoneInput extends Component {
  state = {
    isOpen: false,
    suggestion: '',
    suggestions: [],
    error: '',
    numberInfo: {}
  };

  componentDidMount = () => {
    const { value } = this.props;
    if (value.length > 3) this.parseNumber(value);
  };

  handleSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      suggestions: getSuggestions(value)
    });
  };

  handleSuggestionsClearRequested = () => {
    this.setState({
      suggestions: []
    });
  };

  onSuggestionSelected = (event, { suggestionValue }) => {
    const { onChange } = this.props;
    const { numberInfo } = this.state;
    const country = this.findCountryByName(suggestionValue);
    let newVal = '';
    if (country) {
      newVal = `${country.dialCode}`;
      if (numberInfo) {
        newVal += `${numberInfo.nationalNumber}`;
      }
      onChange(newVal);
      this.setState({ numberInfo: this.parseNumber(newVal) });
    }
  };

  handleSuggestionChange = (event, { newValue }) => {
    this.setState({
      suggestion: newValue
    });
  };

  toggleOpen = event => {
    const { currentTarget } = event;
    this.setState(state => ({ anchorEl: currentTarget, isOpen: !state.isOpen }));
  };

  handleInputChange = event => {
    // console.log('handle input change');
    const { onChange } = this.props;
    const { value } = event.target;
    onChange(value);
    this.parseNumber(value);
  };

  parseNumber = value => {
    try {
      const { metadata, ...numberInfo } = parsePhoneNumber(value);
      this.setState({ error: '', numberInfo });
      return numberInfo;
    } catch (error) {
      this.setState({ error: errors[`${error.message}`] });
    }
  };

  findCountryByName = name => suggestionsLists.find(c => c.label.toLowerCase() === name.toLowerCase());

  render() {
    const { isOpen, suggestion, suggestions, anchorEl, error, numberInfo } = this.state;
    const { classes, label, value, defaultCountry, textFieldProps } = this.props;
    const autosuggestProps = {
      renderInputComponent,
      suggestions,
      onSuggestionsFetchRequested: this.handleSuggestionsFetchRequested,
      onSuggestionsClearRequested: this.handleSuggestionsClearRequested,
      getSuggestionValue,
      renderSuggestion
    };

    return (
      <React.Fragment>
        <TextField
          onChange={this.handleInputChange}
          value={value}
          error={Boolean(error)}
          helperText={Boolean(error) && error}
          InputProps={{
            startAdornment: (
              <InputAdornment position="start">
                <Button size="small" style={{ paddingTop: 0, paddingBottom: 0 }} onClick={this.toggleOpen}>
                  <FlagIcon
                    code={numberInfo.country ? numberInfo.country.toLowerCase() : defaultCountry.toLowerCase()}
                  />
                  <ArrowDropDown />
                </Button>
              </InputAdornment>
            )
          }}
          {...textFieldProps}
        />
        <Popper open={isOpen} transition style={{ width: 'auto' }} anchorEl={anchorEl}>
          {({ TransitionProps }) => (
            <Fade {...TransitionProps} timeout={350}>
              <Paper elevation={1}>
                <Autosuggest
                  focusInputOnSuggestionClick={false}
                  onSuggestionSelected={this.onSuggestionSelected}
                  {...autosuggestProps}
                  inputProps={{
                    classes,
                    placeholder: 'Type to search...',
                    value: suggestion,
                    onChange: this.handleSuggestionChange,
                    onBlur: this.toggleOpen,
                    inputRef: node => {
                      this.popperNode = node;
                    },
                    InputLabelProps: {
                      shrink: true
                    }
                  }}
                  theme={{
                    suggestionsList: classes.suggestionsList,
                    suggestion: classes.suggestion
                  }}
                  renderSuggestionsContainer={options => (
                    <Paper square {...options.containerProps} elevation={1}>
                      {options.children}
                    </Paper>
                  )}
                />
              </Paper>
            </Fade>
          )}
        </Popper>
      </React.Fragment>
    );
  }
}
export default withStyles(styles)(PhoneInput);
