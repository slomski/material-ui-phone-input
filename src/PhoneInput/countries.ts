import countryNames from './countries';
import { ICountry } from './PhoneInput';
const countries: ICountry[] = [];

countryNames.forEach(country => {
  countries.push(country);
});

export default countries;
