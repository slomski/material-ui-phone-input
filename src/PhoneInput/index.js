import PhoneInput from './PhoneInput';

// eslint-disable-next-line
window.__MUI_USE_NEXT_TYPOGRAPHY_VARIANTS__ = true;

export default PhoneInput;
