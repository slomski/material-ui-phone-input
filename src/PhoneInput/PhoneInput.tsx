import * as React from 'react';
import { withStyles, StyledComponentProps, createStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField, { TextFieldProps } from '@material-ui/core/TextField';
import Input from '@material-ui/core/Input';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
import Paper from '@material-ui/core/Paper';
import Popper from '@material-ui/core/Popper';
import { CountryCode, CountryCallingCode, NationalNumber, Extension, NumberType, PhoneNumber } from 'libphonenumber-js';
import { parsePhoneNumber } from 'libphonenumber-js/custom';
import * as metadata from 'libphonenumber-js/metadata.min.json';
import Fade from '@material-ui/core/Fade';
import countries from './countries';

export interface ICountry {
  name: string;
  dialCode: string;
  code: string;
}

export interface ISuggestion {
  value: string;
  label: string;
  dialCode: string;
}

export enum errors {
  TOO_SHORT = 'Number is too short',
  TOO_LONG = 'Number is too long',
  INVALID_COUNTRY = 'Country code in number is invalid',
  NOT_A_NUMBER = 'Invalid number provided',
  INVALID_NUMBER = 'Number provided is not valid',
}

export interface IPhoneNumber extends PhoneNumber {
  metadata?: object;
}

interface IPhoneInputNumberInfo {
  countryCallingCode: CountryCallingCode;
  country?: CountryCode;
  nationalNumber: NationalNumber;
  number: string;
  carrierCode?: string;
  ext?: Extension;
}

interface IPhoneInputState {
  isOpen: boolean;
  suggestion: string;
  suggestions: ISuggestion[];
  error: string;
  numberInfo: IPhoneInputNumberInfo | undefined;
  anchorEl: EventTarget & Element | null;
}

interface IPhoneNumberInfoProps extends StyledComponentProps {
  value: string;
  onChange: (value: string) => void;
  metadata: object;
  textFieldProps?: TextFieldProps;
  validationType?: NumberType[];
  onValidationSuccess?: (res: any) => void;
  onValidationFailed?: (res: any) => void;
}

const suggestionsLists: ISuggestion[] = countries.map(option => ({
  value: option.code,
  label: option.name,
  dialCode: option.dialCode,
}));

const styles = createStyles({
  suggestion: {
    display: 'block',
  },
  suggestionsList: {
    margin: 0,
    padding: 0,
    listStyleType: 'none',
  },
  input: {
    padding: 10,
    border: 0,
    width: '100%',
  },
});

function getSuggestions(country: string) {
  const inputValue = country.trim().toLowerCase();
  const inputLength = country.length;
  let count = 0;

  return inputLength === 0
    ? []
    : suggestionsLists.filter(suggestion => {
        const keep = count < 5 && suggestion.label.slice(0, inputLength).toLowerCase() === inputValue;
        if (keep) {
          count += 1;
        }
        return keep;
      });
}

function flagIcon(code: string): React.ReactNode {
  return <img src={require(`../flags/${code}.png`)} />;
}

class PhoneInput extends React.Component<IPhoneNumberInfoProps, IPhoneInputState> {
  public static defaultProps: Partial<IPhoneNumberInfoProps> = {
    metadata: metadata,
  };
  popperNode: HTMLElement | null = null;
  state = {
    isOpen: false,
    suggestion: '',
    suggestions: [],
    error: '',
    anchorEl: null,
    numberInfo: {
      countryCallingCode: '',
      nationalNumber: '',
      number: '',
      country: '001' as CountryCode,
      phone: '',
      isPossible: false,
      isValid: false,
    },
  };

  componentDidMount = () => {
    const { value } = this.props;
    if (value.length > 3) {
      const numberInfo = this.parseNumber(value);
      this.matchSuggestion(numberInfo);
    }
  };

  handleSuggestionsFetchRequested = (value: string) => {
    this.setState({
      suggestions: getSuggestions(value),
    });
  };

  clearSuggestions = () => {
    this.setState({
      suggestions: [],
    });
  };

  matchSuggestion(numberInfo: IPhoneNumber | undefined): void {
    if (numberInfo && numberInfo.country) {
      const country = numberInfo.country.toLowerCase();
      const suggestion = suggestionsLists.find(e => e.value.toLocaleLowerCase() === country);
      this.setState({
        suggestion: suggestion ? suggestion.label : '',
      });
    }
  }

  onSuggestionSelected = (event: React.SyntheticEvent, suggestion: string) => {
    const { onChange } = this.props;
    const { numberInfo } = this.state;
    const country = this.findCountryByName(suggestion);
    let newVal = '';
    if (country) {
      newVal = `${country.dialCode}`;
      if (numberInfo) {
        newVal += `${numberInfo.nationalNumber}`;
      }
      onChange(newVal);
      this.setState({ numberInfo: this.parseNumber(newVal), suggestion, suggestions: [] });
    }
  };

  handleSuggestionChange = (event: React.SyntheticEvent) => {
    const { value } = event.target as HTMLInputElement;
    this.setState({
      suggestion: value,
    });
    this.handleSuggestionsFetchRequested(value);
  };

  toggleOpen = (event: React.SyntheticEvent) => {
    const { currentTarget } = event;
    this.setState({ anchorEl: currentTarget, isOpen: !this.state.isOpen });
  };

  handleInputChange = (event: React.SyntheticEvent) => {
    const { onChange } = this.props;
    const { value } = event.target as HTMLInputElement;
    onChange(value);
    this.parseNumber(value);
  };

  parseNumber = (value: string): IPhoneNumber | undefined => {
    const { metadata, validationType, onValidationSuccess, onValidationFailed } = this.props;
    let phoneNumber: IPhoneNumber;
    try {
      phoneNumber = parsePhoneNumber(value, metadata);
      const { metadata: meta, ...rest } = phoneNumber;
      const response = {
        ...rest,
        type: phoneNumber.getType(),
        isValid: phoneNumber.isValid(),
        isPossible: phoneNumber.isPossible(),
      };
      if (validationType) {
        if (validationType.includes(phoneNumber.getType()) && phoneNumber.isValid()) {
          onValidationSuccess && onValidationSuccess(response);
          const suggestion = this.findSuggestionByCode(phoneNumber.country);
          console.log(suggestion);
          this.setState({
            error: '',
            numberInfo: phoneNumber,
            suggestion,
          });
        } else {
          onValidationFailed && onValidationFailed(response);
          this.setState({
            error: errors.INVALID_NUMBER,
            numberInfo: phoneNumber,
            suggestion: this.findSuggestionByCode(phoneNumber.country),
          });
        }
      } else {
        phoneNumber = parsePhoneNumber(value, metadata);
        this.setState({ error: '', numberInfo: phoneNumber });
      }
      return phoneNumber;
    } catch (error) {
      this.setState({ error: errors[`${error.message}`] });
      return;
    }
  };

  findCountryByName = (name: string) => suggestionsLists.find(c => c.label.toLowerCase() === name.toLowerCase());

  findSuggestionByCode = (code: CountryCode | undefined) => {
    if (code) {
      const suggestion = suggestionsLists.find(c => c.value.toLowerCase() === code.toLowerCase());
      return (suggestion && suggestion.label) || '';
    }
    return '';
  };

  handleSuggestionBlur = () => {
    this.setState({ isOpen: false });
  };

  render() {
    const { error, numberInfo, isOpen, suggestion, suggestions } = this.state;
    const { value, textFieldProps } = this.props;
    const suggestionsToRender = suggestions.map((s: ISuggestion) => {
      return (
        <ListItem button onClick={e => this.onSuggestionSelected(e, s.label)} key={s.value}>
          <ListItemText primary={s.label} />
        </ListItem>
      );
    });
    return (
      <div
        ref={node => {
          this.popperNode = node;
        }}
      >
        <TextField
          onChange={this.handleInputChange}
          fullWidth
          value={value}
          error={Boolean(error)}
          helperText={Boolean(error) && error}
          InputProps={{
            startAdornment: (
              <InputAdornment position="start">
                <Button size="small" style={{ paddingTop: 0, paddingBottom: 0 }} onClick={this.toggleOpen}>
                  {flagIcon(numberInfo && numberInfo.country ? numberInfo.country.toLowerCase() : '001')}
                  <ArrowDropDown />
                </Button>
              </InputAdornment>
            ),
          }}
          {...textFieldProps}
        />
        <Popper open={isOpen} transition disablePortal anchorEl={this.popperNode} placement="bottom-start">
          {({ TransitionProps }) => (
            <Fade {...TransitionProps} timeout={350}>
              <Paper elevation={1} style={{ width: this.popperNode ? this.popperNode.clientWidth : undefined }}>
                <Input
                  autoFocus
                  fullWidth
                  disableUnderline
                  style={{ padding: '5px 10px 0px 10px' }}
                  onBlur={this.handleSuggestionBlur}
                  onChange={this.handleSuggestionChange}
                  value={suggestion}
                />
                {suggestionsToRender && (
                  <div>
                    <Divider />
                    <List component="nav" style={{ paddingTop: '0px', paddingBottom: '0px' }}>
                      {suggestionsToRender}
                    </List>
                  </div>
                )}
              </Paper>
            </Fade>
          )}
        </Popper>
      </div>
    );
  }
}
export default withStyles(styles)(PhoneInput);
