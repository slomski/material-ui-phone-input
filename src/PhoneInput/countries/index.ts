import * as countryNames from './countries.json';
import { ICountry } from './types';

const countries: ICountry[] = [];

countryNames.forEach(country => {
  countries.push(country);
});

export default countries;
