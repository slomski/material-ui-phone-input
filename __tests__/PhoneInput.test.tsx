import * as React from 'react';
import PhoneInput from '../src/PhoneInput/PhoneInput';
import * as renderer from 'react-test-renderer';

const handleChange = jest.fn();

it('PhoneInput no value renders correctly', () => {
  const tree = renderer.create(<PhoneInput value="" onChange={handleChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('PhoneInput with value renders correctly', () => {
  const tree = renderer.create(<PhoneInput value="+48501502503" onChange={handleChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});
