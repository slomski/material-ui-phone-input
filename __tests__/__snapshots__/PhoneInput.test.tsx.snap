// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInput no value renders correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root-4 MuiFormControl-fullWidth-7"
  >
    <div
      className="MuiInputBase-root-21 MuiInput-root-8 MuiInput-underline-12 MuiInputBase-fullWidth-30 MuiInput-fullWidth-15 MuiInputBase-formControl-22 MuiInput-formControl-9 MuiInputBase-adornedStart-25"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root-38 MuiInputAdornment-positionStart-40"
      >
        <button
          className="MuiButtonBase-root-68 MuiButton-root-42 MuiButton-text-44 MuiButton-flat-47 MuiButton-sizeSmall-65"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-43"
          >
            <img
              src="test-file-stub"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-71"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-80"
          />
        </button>
      </div>
      <input
        aria-invalid={false}
        className="MuiInputBase-input-31 MuiInput-input-16 MuiInputBase-inputAdornedStart-36"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`PhoneInput with value renders correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root-4 MuiFormControl-fullWidth-7"
  >
    <div
      className="MuiInputBase-root-21 MuiInput-root-8 MuiInput-underline-12 MuiInputBase-fullWidth-30 MuiInput-fullWidth-15 MuiInputBase-formControl-22 MuiInput-formControl-9 MuiInputBase-adornedStart-25"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root-38 MuiInputAdornment-positionStart-40"
      >
        <button
          className="MuiButtonBase-root-68 MuiButton-root-42 MuiButton-text-44 MuiButton-flat-47 MuiButton-sizeSmall-65"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-43"
          >
            <img
              src="test-file-stub"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-71"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-80"
          />
        </button>
      </div>
      <input
        aria-invalid={false}
        className="MuiInputBase-input-31 MuiInput-input-16 MuiInputBase-inputAdornedStart-36"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="+48501502503"
      />
    </div>
  </div>
</div>
`;
